tests/zkevm/test_worst_compute.py

```diff
@@ -57,6 +58,243 @@ def make_dup(index: int) -> Opcode:
+@pytest.mark.parametrize("non_zero_value", [True, False])
+@pytest.mark.parametrize("from_origin", [True, False])
+def test_worst_callvalue(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    non_zero_value: bool,
+    from_origin: bool,
+):
+    """
+    Test running a block with as many CALLVALUE opcodes as possible.
+
+    The `non_zero_value` parameter controls whether opcode must return non-zero value.
+    The `from_origin` parameter controls whether the call frame is the immediate from the
+    transaction or a previous CALL.
+    """
+    env = Environment()
+
+    code_prefix = Op.JUMPDEST
+    iter_loop = Op.POP(Op.CALLVALUE)
+    code_suffix = Op.PUSH0 + Op.JUMP
+    code_iter_len = (MAX_CODE_SIZE - len(code_prefix) - len(code_suffix)) // len(iter_loop)
+    code = code_prefix + iter_loop * code_iter_len + code_suffix
+    assert len(code) <= MAX_CODE_SIZE
+    code_address = pre.deploy_contract(code=bytes(code))
+
+    tx_to = (
+        code_address
+        if from_origin
+        else pre.deploy_contract(
+            code=Op.CALL(address=code_address, value=1 if non_zero_value else 0), balance=10
+        )
+    )
+
+    tx = Transaction(
+        to=tx_to,
+        gas_limit=env.gas_limit,
+        value=1 if non_zero_value and from_origin else 0,
+        sender=pre.fund_eoa(),
+    )
+
+    state_test(
+        env=env,
+        pre=pre,
+        post={},
+        tx=tx,
+    )
+
+
+class ReturnDataStyle(Enum):
+    """Helper enum to specify return data is returned to the caller."""
+
+    RETURN = auto()
+    REVERT = auto()
+    IDENTITY = auto()
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize("non_zero_size", [True, False])
+@pytest.mark.parametrize(
+    "return_data_style",
+    [
+        ReturnDataStyle.RETURN,
+        ReturnDataStyle.REVERT,
+        ReturnDataStyle.IDENTITY,
+    ],
+)
+def test_worst_returndatasize(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    non_zero_size: bool,
+    return_data_style: ReturnDataStyle,
+):
+    """
+    Test running a block with as many RETURNDATASIZE opcodes as possible.
+
+    The `non_zero_size` parameter controls whether opcode must return non-zero value.
+    The `return_data_style` indicates how returned data is produced for the opcode caller.
+    """
+    env = Environment()
+
+    dummy_contract_call = Bytecode()
+    if non_zero_size:
+        if return_data_style == ReturnDataStyle.IDENTITY:
+            dummy_contract_call = Op.STATICCALL(
+                address=pre.deploy_contract(
+                    code=Op.REVERT(0, 1)
+                    if return_data_style == ReturnDataStyle.REVERT
+                    else Op.RETURN(0, 1)
+                )
+            )
+        else:
+            dummy_contract_call = Op.MSTORE8(0, 1) + Op.STATICCALL(
+                address=0x04,  # Identity precompile
+                args_size=1,
+            )
+
+        code_prefix = dummy_contract_call + Op.JUMPDEST
```

Comment:
This test now produces 3 similar tests if `non_zero_size` = `False`. I think that also for RETURN/REVERT/IDENTITY we want to return also the empty buffer to see if this makes any difference

Reply:
Not sure why this is marked as outdated, but this applies to the RETURNDATASIZE test

Reply:
I'll split the test I think to avoid this dimension blowup of `non_zero_size = False` and leveraging that include that "return 0" case!